lib/core/update/check_github_update.dart

```diff
@@ -37,3 +37,20 @@ Future<Version> fetchVersion() async {
     return Version(version: 'N/A', latestVersion: 'N/A', hasUpdate: false);
   }
 }
+
+String _trimV(String version) {
+  if (version.startsWith('v')) return version.substring(1);
+  return version;
+}
+
+/// Returns true if [second] is greater than [first]
+bool _compareVersions(version_parser.Version first, version_parser.Version second) {
+  // Handle different builds (which are not normally part of semver sorting)
+  int? firstBuild = int.tryParse(first.build);
+  int? secondBuild = int.tryParse(second.build);
+  if (firstBuild != null && secondBuild != null && firstBuild != secondBuild) {
+    return secondBuild > firstBuild;
+  }
+
+  return second > first;
+}
```

Comment:
Despite using the `version` package, we still have to do some custom handling because our versioning is not purely semver in terms of sorting. Here are the scenarios I tested. In each case, the second should be considered greater than the first.

* Alpha to alpha across releases
  * 0.2.1-alpha.2+13 -> 0.2.1-alpha.1+14
* Alpha to alpha within a release
  * 0.2.1-alpha.0+14 -> 0.2.1-alpha.1+14
* Alpha to release
  * 0.2.1-alpha.4+13 -> 0.2.1+13
* Release to alpha
  * 0.2.1+13 -> 0.2.1-alpha.1+14